projecttacoma/deqm-test-server-frontend
# Evaluate measure date inputs

## Summary
Implementation of pre-filled date pickers for the Evaluate Measure page. 
## New behavior
On the Evaluate Measure page for a valid Measure, two DatePickers are rendered, for choosing period start and period end for the evaluate measure. 
- If the Measure has dates in its effectivePeriod, those are used to pre-fill the DatePickers. 
- If no dates are specified, the DatePickers are pre-filled with default values of January 1, 2022 and December 31, 2022.
- If a user attempts to navigate to Measure/[invalid-id]/evaluate then an error notification appears on the screen.
## Code changes
- **pages/[resource]/[id]/evaluate.tsx**  renders DatePickers/error notif 
- **__tests__/pages/resource/id/evaluate.test.tsx**, unit tests added to file
-  minor changes to package.json, tests/pages/resource/id/update.test.tsx

## Testing Guidance
See the README.md for first time setup instructions.
#### Testing in browser: 
- To start the frontend: `npm run dev` then navigate to http://localhost:3001 in a browser
- Click on “Measure” from the left-hand-side navigation menu. Click on any of the IDs. Click on the “Evaluate Measure” button near the top right of the main page body.
- Valid Measure:
  - Two DatePickers should be rendered on the Evaluate Measure page. They should have the same values as the periodStart and periodEnd of the Measure. If those value are not specified in the Measure, the DatePickers should have the values: January 1, 2022  and December 31, 2022.
  - User should be able to type date into the DatePickers, or choose a date using the DatePicker’s calendar.
- Invalid Measure:
  - change the URL to have a Measure ID that does not exist. An error notification should appear specifying the issue.
#### Unit testing: 
- Run the unit tests: `npm run test`
